### PR TITLE
Modification about random combine

### PR DESCRIPTION
**Pull request by yaozengwei**
This PR modifies the random combining of different encoder layers. 
* It combines from 1/3 layers, i.e., layer-4 (counts from 0).
* It removes linear layers in `RandomCombine` class. 

Experimental results can be found on https://github.com/k2-fsa/icefall/issues/431.

### CODE REVIEW

File: egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py [hunk 1]
```diff
@@ -1161,23 +1150,8 @@ def forward(self, inputs: List[Tensor]) -> Tensor:
...
-            #      mod_inputs.append(self.linear[i](inputs[i]))
-            assert False
-        else:
-            for i, linear in enumerate(self.linear):
-                if i < num_inputs - 1:
-                    mod_inputs.append(linear(inputs[i]))
-
-        mod_inputs.append(inputs[num_inputs - 1])
+        for i in range(num_inputs):
+            mod_inputs.append(inputs[i])
```
Comment (pkufool) [inline egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py:1154]: Do we need `mod_inputs` any more?

**Comment by danpovey** [comment]
We have to watch out, here, because this would invalidate already-trained models.  If we have uploaded a trained model from this recipe, I suggest we could re-upload a new one.

**Comment by yaozengwei** [comment]
> We have to watch out, here, because this would invalidate already-trained models. If we have uploaded a trained model from this recipe, I suggest we could re-upload a new one.



> We have to watch out, here, because this would invalidate already-trained models. If we have uploaded a trained model from this recipe, I suggest we could re-upload a new one.

OK, I will re-upload a new one.

**Comment by csukuangfj** [comment]
> OK, I will re-upload a new one.

Have you finished uploading the pre-trained models? Please also update [RESULTS.md](https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/RESULTS.md)

File: egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py [hunk 2]
```diff
@@ -1072,7 +1077,6 @@ class RandomCombine(nn.Module):
     def __init__(
         self,
         num_inputs: int,
-        num_channels: int,
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py:1075]: please also update the test code
https://github.com/k2-fsa/icefall/blob/8761452a2c6f5fba0617f0172ede4019e7b83b13/egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py#L1270

Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py:1075]: Please also remove the following doc:
https://github.com/k2-fsa/icefall/blob/8761452a2c6f5fba0617f0172ede4019e7b83b13/egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py#L1067-L1069

**Comment by yaozengwei** [comment]
> > OK, I will re-upload a new one.
> 
> Have you finished uploading the pre-trained models? Please also update [RESULTS.md](https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/RESULTS.md)

Thanks for the reminder. See https://github.com/k2-fsa/icefall/pull/467.

**Pull request merged**